src/lidar_subscriber.cpp

```diff
@@ -0,0 +1,156 @@
+
+#include <memory>
+#include <cmath>
+#include "rclcpp/rclcpp.hpp"
+
+#include "core-structs.h"
+
+
+#include "sensor_msgs/msg/laser_scan.hpp"
+#include <queue>
+#include <limits>
+#include "visualization_msgs/msg/marker.hpp"
+
+constexpr float LANDMARK_THRESHOLD_m = 0.02;
+constexpr bool testing = true;
+
+bool same_landmark(const Observation2D &first, const Observation2D &second){
+  float r1 = first.range_m;
+  float r2 = second.range_m;
+  float theta1 = first.bearing_rad;
+  float theta2 = second.bearing_rad;
+  float distance = sqrt(pow(r1,2.0)+pow(r2, 2.0)-2*r1*r2*cos(theta1-theta2));
+  if (distance >= LANDMARK_THRESHOLD_m){
+	//std::cout<<distance<<" ";		 
+   }
+  return  distance <= LANDMARK_THRESHOLD_m;
+}
+
+Observation2D merge(Observation2D &existing_landmark, Observation2D &new_measurement, int &count){
```

Comment:
So should the overloaded functions here be static.

Reply:
Done!